src/graph/src/lockfile/load-edges.ts

```diff
@@ -48,13 +48,18 @@ export const loadEdges = (
         },
       })
     }
+
+    // sets a registry for this spec to inherit from
+    spec.inheritedRegistry = from.registry
+
```

Comment:
implement in the lockfile logic the same logic present in the actual load module